test/SignerBondsUniswapV2.test.js

```diff
@@ -0,0 +1,251 @@
+        ethers.utils.parseEther("10")
+      )
+    })
+  })
+
+  describe("setMaxAllowedPriceImpact", () => {
+    context("when the caller is not the governance", () => {
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2.connect(other).setMaxAllowedPriceImpact("150")
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called with value bigger than max", () => {
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2
+            .connect(governance)
+            .setMaxAllowedPriceImpact("10001")
+        ).to.be.revertedWith("Maximum value is 10000 basis points")
+      })
+    })
+
+    context("when the caller is the governance", () => {
+      it("should set max allowed price impact parameter", async () => {
+        await signerBondsUniswapV2
+          .connect(governance)
+          .setMaxAllowedPriceImpact("150")
+        expect(await signerBondsUniswapV2.maxAllowedPriceImpact()).to.be.equal(
+          "150"
+        )
+      })
+    })
+  })
+
+  describe("setSlippageTolerance", () => {
+    context("when the caller is not the governance", () => {
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2.connect(other).setSlippageTolerance("100")
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called with value bigger than max", () => {
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2.connect(governance).setSlippageTolerance("10001")
+        ).to.be.revertedWith("Maximum value is 10000 basis points")
+      })
+    })
+
+    context("when the caller is the governance", () => {
+      it("should set slippage tolerance parameter", async () => {
+        await signerBondsUniswapV2
+          .connect(governance)
+          .setSlippageTolerance("100")
+        expect(await signerBondsUniswapV2.slippageTolerance()).to.be.equal(
+          "100"
+        )
+      })
+    })
+  })
+
+  describe("setSwapDeadline", () => {
+    context("when the caller is not the governance", () => {
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2.connect(other).setSwapDeadline("2400")
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the governance", () => {
+      it("should set slippage tolerance parameter", async () => {
+        await signerBondsUniswapV2.connect(governance).setSwapDeadline("2400")
+        expect(await signerBondsUniswapV2.swapDeadline()).to.be.equal("2400")
+      })
+    })
+  })
+
+  describe("swapSignerBondsOnUniswapV2", () => {
+    const ethReserves = 1000
+    const tokenReserves = 5000
+    const exchangeRate = 5 // because one can get 5 tokens for every 1 ETH
+
+    beforeEach(async () => {
+      await uniswapV2PairStub.setReserves(
```

Comment:
If you're fine with the other comment regarding mocking `uniswapV2Pair`, then this can be replaced by the mock.

Reply:
https://github.com/keep-network/coverage-pools/pull/66/commits/c828b33c022f6043dbd98f812e8a44f7e058b75c